dome9/data_source_dome9_admission_control_policy.go

```diff
@@ -0,0 +1,64 @@
+package dome9
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"log"
+)
+
+func dataSourceAdmissionControlPolicy() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAdmissionControlPolicyRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"target_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"target_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ruleset_id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"notification_ids": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"action": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAdmissionControlPolicyRead(d *schema.ResourceData, meta interface{}) error {
+	d9Client := meta.(*Client)
+
+	policyID := d.Get("id").(string)
+	log.Printf("Getting data for Admission Control Policy id: %s\n", policyID)
+
+	resp, _, err := d9Client.admissionControlPolicy.Get(policyID)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(resp.ID)
```

Comment:
let me try if d.SetId is enough, since as understood from the documentation, SetId will set the id for the terraform resource, let me check it and see that we still get the id while reading the resource

Reply:
i see it is enough, testing it and found that the tf resource holds the same policy id